src/core/qgsvectorfilewriter.cpp

```diff
@@ -982,6 +982,48 @@ QMap<QString, QgsVectorFileWriter::MetaData> QgsVectorFileWriter::initMetaData()
                          )
                        );
 
```

Comment:
Or QGIS could possibly have a small script that would dynamically iterate through OGR drivers and extract their options.